pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java

```diff
@@ -51,10 +51,10 @@ public DefaultExtensionFinder(PluginManager pluginManager, ExtensionFactory exte
         }
 
 		log.debug("Finding extensions for extension point '{}'", type.getName());
-        readIndexFiles();
+        Map<String, Set<String>> cachedEntries = readIndexFiles();
```

Comment:
Can you do a rebase please. I modified the class a little bit to improve the testability.